src/useStepState/index.js

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from 'react';
+import { throwError } from '@clutch/helpers';
+
+export const useStepState = ({
+  steps = [],
+  initialStepIndex = 0,
+  useStateDep = useState,
+  useEffectDep = useEffect,
+}) => {
+  const [currentStepIndexState, setCurrentStepIndexState] = useStateDep(initialStepIndex);
+  const [currentStepState, setCurrentStepState] = useStateDep(steps[initialStepIndex]);
+
+  !steps[currentStepIndexState] && throwError('You are trying to go to a step that does not exist');
```

Comment:
```suggestion
  !steps[initialStepIndex] && throwError('You are trying to go to a step that does not exist');
```
And move this to top so the useStates aren't called